internal/engine/compiler/impl_amd64_test.go

```diff
@@ -391,6 +391,42 @@ func TestAmd64Compiler_readInstructionAddress(t *testing.T) {
 	})
 }
 
+func TestAmd64Compiler_preventCrossedTargetdRegisters(t *testing.T) {
+	env := newCompilerEnvironment()
+	compiler := env.requireNewCompiler(t, newAmd64Compiler, nil).(*amd64Compiler)
+
+	rv := []*runtimeValueLocation{{register: amd64.RegAX}, {register: amd64.RegCX}, {register: amd64.RegDX}}
+
+	currentRegisters := func() []asm.Register {
```

Comment:
I don't think we need this as I see this just adds the complexity.

Reply:
I'll move it to a separate function then, because I need to collect registers three times

Reply:
cool